libraries/AP_Follow/AP_Follow.cpp

```diff
@@ -320,7 +322,75 @@ void AP_Follow::handle_msg(const mavlink_message_t &msg)
             _sysid.set(msg.sysid);
             _automatic_sysid = true;
         }
+        updated = true;
+        break;
+    }
+    case MAVLINK_MSG_ID_FOLLOW_TARGET: {
+        // get estimated location and velocity (for logging)
+        Vector3f vel_estimate;
+        UNUSED_RESULT(get_target_location_and_velocity(loc_estimate, vel_estimate));
+
+        // decode message
+        mavlink_follow_target_t packet;
+        mavlink_msg_follow_target_decode(&msg, &packet);
 
+        // ignore message if lat and lon are (exactly) zero
+        if ((packet.lat == 0 && packet.lon == 0)) {
+            return;
+        }
+        // require at least position
+        if ((packet.est_capabilities & (1<<0)) == 0) {
+            return;
+        }
+
+        _target_location.lat = packet.lat;
+        _target_location.lng = packet.lon;
+
+        float alt = packet.alt;
+        // select altitude source based on FOLL_ALT_TYPE param 
+        if (_alt_type == AP_FOLLOW_ALTITUDE_TYPE_RELATIVE) {
+            // convert to relative to origin
+            Location origin;
+            if (!AP::ahrs().get_origin(origin)) {
+                return;
+            }
+            alt -= origin.alt * 0.01;
```

Comment:
I'm not sure this is correct.  "relative" should be relative to home but this appears to be relative to the EKF origin.

Reply:
fixed